### PR TITLE
Add email templates

### PR DESCRIPTION
**Pull request by stepanov**

### CODE REVIEW

File: src/Template/Element/Email/activation_link.ctp [hunk 1]
```diff
@@ -0,0 +1,30 @@
+<!-- Body content -->
+<tr>
+    <td class="content-cell">
+        <h1><?= __("Hi {0}", !empty($first_name) ? $first_name : $username); ?>,</h1>
+        <p>Please activate your <?= getenv('PROJECT_NAME') ?> account. Use the button below to do that. <strong>This activation link is only valid for the next 24 hours.</strong></p>
```
Comment (mamchenkov) [inline src/Template/Element/Email/activation_link.ctp]: @stepanov `PROJECT_NAME` variable is not always defined.  We have some trickery around footer and System Information screens to populate it, but it needs to be reusable.  I'll get back to you on this one.

File: src/Template/Element/Email/activation_link.ctp [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+            <tr>
+                <td align="center">
+                <!-- Border based button https://litmus.com/blog/a-guide-to-bulletproof-buttons-in-email-design -->
+                    <table width="100%" border="0" cellspacing="0" cellpadding="0">
+                        <tr>
+                            <td align="center">
+                                <table border="0" cellspacing="0" cellpadding="0">
+                                    <tr>
+                                        <td>
+                                            <a href="<?= $this->Url->build($activationUrl); ?>" class="button button--green" target="_blank">Activate your account</a>
```
Comment (mamchenkov) [inline src/Template/Element/Email/activation_link.ctp]: @stepanov I think it's better if we push all the necessary variables from the controller, as these can be executed both via the web request and outside (shell/cron/etc).

File: src/Template/Element/Email/footer.ctp [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+                                    </tr>
+                                </table>
+                            </td>
+                        </tr>
+                        <tr>
+                            <td>
+                                <table class="email-footer" align="center" width="570" cellpadding="0" cellspacing="0">
+                                    <tr>
+                                        <td class="content-cell" align="center">
+                                            <p class="sub align-center">&copy; 2017 Qobo Ltd. All rights reserved.</p>
```
Comment (mamchenkov) [inline src/Template/Element/Email/footer.ctp]: @stepanov this should not be hardcoded.  I need to find you some configuration variables to fetch this stuff from :)

File: src/Template/Element/Email/header.ctp [hunk 4]
```diff
@@ -0,0 +1,32 @@
+<?php
+use Cake\Core\Configure;
+?>
+
+<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
+<html xmlns="http://www.w3.org/1999/xhtml">
+    <head>
+        <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+        <meta http-equiv="Content-Type" content="text/html; charset=UTF-8" />
+        <title>Set up a new password for [Product Name]</title>
```
Comment (mamchenkov) [inline src/Template/Element/Email/header.ctp]: @stepanov missed placeholder

File: src/Template/Element/Email/activation_link.ctp (hunk 2 above)
Comment (stepanov) [inline src/Template/Element/Email/activation_link.ctp]: @mamchenkov I moved all logic into Helper

File: src/View/Helper/HtmlEmailHelper.php [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+    /**
+     * setVariables method
+     *
+     * @return void
+     */
+    protected function setVariables()
+    {
+        $projectName = getenv('PROJECT_NAME');
+        $projectUrl = getenv('PROJECT_URL');
+        $copyright = date('Y') . ' ' . $projectName . '. All rigghts reserved.';
```
Comment (mamchenkov) [inline src/View/Helper/HtmlEmailHelper.php]: @stepanov 'rights', not 'rigghts'

File: src/View/Helper/HtmlEmailHelper.php [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+        $logo = Configure::read('Theme.logo.mini');
+
+        $this->params = [
+            'projectName' => $projectName,
+            'projectUrl' => $projectUrl,
+            'copyright' => '&copy ' . date('Y') . ' ' . $projectName . '. All rigghts reserved.',
```
Comment (mamchenkov) [inline src/View/Helper/HtmlEmailHelper.php]: @stepanov use the `$copyright` variable.

File: src/View/Helper/HtmlEmailHelper.php [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+            'logo' => $logo,
+            'footerMsg' => 'Qobo LTD'
```
Comment (mamchenkov) [inline src/View/Helper/HtmlEmailHelper.php]: @stepanov use `$projectName` variable

File: src/Template/Plugin/CakeDC/Users/Email/html/validation.ctp [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+
+$activationUrl = [
+    '_full' => true,
+    'plugin' => 'CakeDC/Users',
+    'controller' => 'Users',
+    'action' => 'validateEmail',
+    isset($token) ? $token : ''
+];
+
+$content = $this->element('Email/header');
```
Comment (mamchenkov) [inline src/Template/Plugin/CakeDC/Users/Email/html/validation.ctp]: @stepanov you have these in the configuration

File: src/Template/Plugin/CakeDC/Users/Email/html/social_account_validation.ctp [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+    $activationUrl = [
+        '_full' => true,
+        'plugin' => 'CakeDC/Users',
+        'controller' => 'SocialAccounts',
+        'action' => 'validateAccount',
+        $socialAccount['provider'],
+        $socialAccount['reference'],
+        $socialAccount['token'],
+    ];
+$content = $this->element('Email/header');
```
Comment (mamchenkov) [inline src/Template/Plugin/CakeDC/Users/Email/html/social_account_validation.ctp]: @stepanov you have this in the configuration

File: src/Template/Element/Email/activation_link.ctp [hunk 10]
```diff
@@ -0,0 +1,30 @@
+<!-- Body content -->
+<tr>
+    <td class="content-cell">
+        <h1><?= __("Hi {0}", !empty($first_name) ? $first_name : $username); ?>,</h1>
```
Comment (mamchenkov) [inline src/Template/Element/Email/activation_link.ctp]: @stepanov this logic should move into the variables as well.  You can have just `$name` in here or something.

File: src/Template/Element/Email/reset_password.ctp [hunk 11]
```diff
@@ -0,0 +1,31 @@
+<!-- Body content -->
+<tr>
+    <td class="content-cell">
+        <h1><?= __("Hi {0}", !empty($first_name) ? $first_name : $username); ?>,</h1>
```
Comment (mamchenkov) [inline src/Template/Element/Email/reset_password.ctp]: @stepanov this logic should move into the variables as well.  You can have just `$name` in here or something.

File: src/Template/Element/Email/social_account_validation.ctp [hunk 12]
```diff
@@ -0,0 +1,30 @@
+<!-- Body content -->
+<tr>
+    <td class="content-cell">
+        <h1><?= __("Hi {0}", !empty($first_name) ? $first_name : $username); ?>,</h1>
```
Comment (mamchenkov) [inline src/Template/Element/Email/social_account_validation.ctp]: @stepanov this logic should move into the variables as well.  You can have just `$name` in here or something.

File: src/Template/Plugin/CakeDC/Users/Email/text/reset_password.ctp [hunk 13]
```diff
@@ -0,0 +1,25 @@
...
+
+$activationUrl = [
+    '_full' => true,
+    'plugin' => 'CakeDC/Users',
+    'controller' => 'Users',
+    'action' => 'resetPassword',
+    isset($token) ? $token : ''
+];
+?>
+<?= __d('CakeDC/Users', "Hi {0}", !empty($first_name) ? $first_name : $username) ?>,
```
Comment (mamchenkov) [inline src/Template/Plugin/CakeDC/Users/Email/text/reset_password.ctp]: @stepanov this logic should move into the variables as well.  You can have just `$name` in here or something.

File: src/Template/Plugin/CakeDC/Users/Email/text/social_account_validation.ctp [hunk 14]
```diff
@@ -0,0 +1,27 @@
...
+        '_full' => true,
+        'plugin' => 'CakeDC/Users',
+        'controller' => 'SocialAccounts',
+        'action' => 'validateAccount',
+        $socialAccount['provider'],
+        $socialAccount['reference'],
+        $socialAccount['token'],
+    ];
+?>
+<?= __d('CakeDC/Users', "Hi {0}", $user['first_name']) ?>,
```
Comment (mamchenkov) [inline src/Template/Plugin/CakeDC/Users/Email/text/social_account_validation.ctp]: @stepanov this logic should move into the variables as well.  You can have just `$name` in here or something.

File: src/Template/Plugin/CakeDC/Users/Email/text/validation.ctp [hunk 15]
```diff
@@ -0,0 +1,25 @@
...
+
+$activationUrl = [
+    '_full' => true,
+    'plugin' => 'CakeDC/Users',
+    'controller' => 'Users',
+    'action' => 'validateEmail',
+    isset($token) ? $token : ''
+];
+?>
+<?= __d('CakeDC/Users', "Hi {0}", isset($first_name)? $first_name : '') ?>,
```
Comment (mamchenkov) [inline src/Template/Plugin/CakeDC/Users/Email/text/validation.ctp]: @stepanov this logic should move into the variables as well.  You can have just `$name` in here or something.

**Pull request merged**